### PR TITLE
Vin/vending v3 upgrade/interface change

### PR DESCRIPTION
**Pull request by SwargamAvinash**
We can Discuss In these PR's

### CODE REVIEW

File: vendingMachine/vendingMachineV3/src/interfaces/UserFunction.java [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package interfaces;
 
-public interface UserVendingMachine {
+public interface UserFunction {
   void insertCoinForPayment();
   void selectProduct();
   void display();
```
Comment (SwargamAvinash) [inline vendingMachine/vendingMachineV3/src/interfaces/UserFunction.java]: Display is not User function I think. I Feel Like it should not be here @vineethreddyp

File: vendingMachine/vendingMachineV3/src/interfaces/VendingMachine.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
 package interfaces;
 
-public interface VendingMachine extends AdminVendingMachine, UserVendingMachine {
+public interface VendingMachine extends UserFunction {
```
Comment (SwargamAvinash) [inline vendingMachine/vendingMachineV3/src/interfaces/VendingMachine.java]: Still Feels a little weird. Okasari State Design Pattern vadi new version rayi you will understand. I feel It is not decoupled ani anipistundhi

File: vendingMachine/vendingMachineV3/src/interfaces/AdminFunction.java [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
 
+
 import entity.Coin;
-import entity.InventoryManagement;
 import entity.Product;
 
-public interface AdminVendingMachine {
+public interface AdminFunction {
   void addQuantityForAProduct(Product product, Integer quantity);
   void printCoinsInMachine();
```
Comment (SwargamAvinash) [inline vendingMachine/vendingMachineV3/src/interfaces/AdminFunction.java]: Admin Functionality Antava Idhi?

File: vendingMachine/vendingMachineV3/src/interfaces/UserFunction.java (hunk 1 above)
Comment (vineethreddyp) [reply vendingMachine/vendingMachineV3/src/interfaces/UserFunction.java]: Yep. makes sense. Will move this to Display Panel

File: vendingMachine/vendingMachineV3/src/entity/DisplayPanel.java [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-import interfaces.UserVendingMachine;
+import interfaces.VendingMachine;
 
 public class DisplayPanel implements DisplayPanelInterface {
 
-  private UserVendingMachine vendingMachine;
+  private VendingMachine vendingMachine;
 
-  public DisplayPanel(UserVendingMachine vendingMachine) {
+  public DisplayPanel(VendingMachine vendingMachine) {
```
Comment (vineethreddyp) [inline vendingMachine/vendingMachineV3/src/entity/DisplayPanel.java]: will be adding display here

**Comment by SwargamAvinash** [comment]
Looks good running kuda bagundhi. Ippudu Idhe danni try to  do it in state design pattern

And Current dantlo you could add some timer also you vending machine right now waits for infinite time until the user gives money -> this is bad

File: vendingMachine/vendingMachineV3/src/impl/AdminImpl.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
+package impl;
+
+
+import entity.Coin;
+import entity.Product;
+import interfaces.Admin;
+import interfaces.CoinManagement;
+import interfaces.ProductManagment;
+
+public class AdminImpl implements Admin {
```
Comment (Bharathgoku) [inline vendingMachine/vendingMachineV3/src/impl/AdminImpl.java:10]: Here admin still can access other functionality like all methods from ProductManagement and CoinManagement, 
even though it's implementing an interface.

Comment (vineethreddyoyo) [reply vendingMachine/vendingMachineV3/src/impl/AdminImpl.java:10]: I need to get both of your views on this. Let discuss this on-call directly @SwargamAvinash

**Review by vineethreddyp — APPROVED** [review]

**Pull request merged**